src/aequitas/bias.py

```diff
@@ -666,59 +711,95 @@ def _get_statistical_significance(cls, original_df, disparity_df, ref_dict,
+                'binary_precision': 'binary_fpr',
+                'binary_tnr': 'binary_fpr',
+                'binary_fdr': 'binary_fpr',
+                'binary_npv': 'binary_fnr',
+                'binary_tpr': 'binary_fnr',
+                'binary_for': 'binary_fnr',
+                'binary_ppr': 'binary_score',
+                'binary_pprev': 'binary_score'}
+
+            # add columns for the rest of columns in dictionary keys
+            # binary score, fnr, fpr already added above
+            for col in (binary_inclusions - binary_col_functions.keys() - {'label_value'}):
+                # binary_col = f'binary_{col}'
+                # original_df.loc[:, binary_col] = original_df.loc[:, SIGNIF_BASES[binary_col]]
+                original_df.loc[:, col] = original_df.loc[:, SIGNIF_BASES[col]]
+
+            # original_df.loc[:, 'binary_precision'] = original_df.loc[:, 'binary_fpr']
+            # original_df.loc[:, 'binary_tnr'] = original_df.loc[:, 'binary_fpr']
+            # original_df.loc[:, 'binary_fdr'] = original_df.loc[:, 'binary_fpr']
+            #
+            # # npv, for, tpr are based on false negatives
+            # original_df.loc[:, 'binary_npv'] = original_df.loc[:, 'binary_fnr']
+            # original_df.loc[:, 'binary_tpr'] = original_df.loc[:, 'binary_fnr']
+            # original_df.loc[:, 'binary_for'] = original_df.loc[:, 'binary_fnr']
+            # # fdr numerator matches fpr numerator
+            #
+            # # pprev and ppr based on score
+            # original_df.loc[:, 'binary_ppr'] = original_df.loc[:, 'binary_score']
+            # original_df.loc[:, 'binary_pprev'] = original_df.loc[:, 'binary_score']
+
+
+            # ensure only predicted positive values included in true positive
+            # error based metrics, and only predicted negative values in false
+            # positive error based metrics
+            POSITIVE_ONLY = ['binary_fpr', 'binary_tnr', 'binary_precision', 'binary_fdr']
+            NEGATIVE_ONLY = ['binary_fnr', 'binary_tpr', 'binary_npv', 'binary_for']
+
+            FPR_BASED = [metric for metric in POSITIVE_ONLY if metric in binary_inclusions]
+            FNR_BASED = [metric for metric in NEGATIVE_ONLY if metric in binary_inclusions]
+
+            original_df.loc[original_df['binary_score'] == 0, FPR_BASED] = pd.np.nan
+            original_df.loc[original_df['binary_score'] == 1, FNR_BASED] = pd.np.nan
 
             measures = list(original_df.columns[original_df.columns.str.contains('binary_')])
             measures += ['label_value']
 
             for measure in measures:
-                cls._calculate_significance(
-                    original_df, disparity_df, attribute, measure,
-                    ref_dict=ref_dict,
-                    alpha=alpha)
+                # only calculate significance if in selected_significance
+                if (measure in selected_significance) or (''.join(measure.split('binary_')) in selected_significance):
```

Comment:
still surprised by this use of join+split – maybe it makes sense, but it seems more cumbersome than necessary.